nixos/modules/rename.nix

```diff
@@ -102,16 +102,47 @@ with lib;
     (mkRenamedOptionModule [ "services" "xserver" "desktopManager" "e19" "enable" ] [ "services" "xserver" "desktopManager" "enlightenment" "enable" ])
 
     # Options that are obsolete and have no replacement.
-    (mkRemovedOptionModule [ "boot" "initrd" "luks" "enable" ])
-    (mkRemovedOptionModule [ "programs" "bash" "enable" ])
-    (mkRemovedOptionModule [ "services" "samba" "defaultShare" ])
-    (mkRemovedOptionModule [ "services" "syslog-ng" "serviceName" ])
-    (mkRemovedOptionModule [ "services" "syslog-ng" "listenToJournal" ])
-    (mkRemovedOptionModule [ "ec2" "metadata" ])
-    (mkRemovedOptionModule [ "services" "openvpn" "enable" ])
-    (mkRemovedOptionModule [ "services" "printing" "cupsFilesConf" ])
-    (mkRemovedOptionModule [ "services" "printing" "cupsdConf" ])
-    (mkRemovedOptionModule [ "services" "xserver" "startGnuPGAgent" ])
+    (mkRemovedOptionModule [ "boot" "initrd" "luks" "enable" ] "")
+    (mkRemovedOptionModule [ "programs" "bash" "enable" ] "")
+    (mkRemovedOptionModule [ "services" "samba" "defaultShare" ] "")
+    (mkRemovedOptionModule [ "services" "syslog-ng" "serviceName" ] "")
+    (mkRemovedOptionModule [ "services" "syslog-ng" "listenToJournal" ] "")
+    (mkRemovedOptionModule [ "ec2" "metadata" ] "")
+    (mkRemovedOptionModule [ "services" "openvpn" "enable" ] "")
+    (mkRemovedOptionModule [ "services" "printing" "cupsFilesConf" ] "")
+    (mkRemovedOptionModule [ "services" "printing" "cupsdConf" ] "")
+    (mkRemovedOptionModule [ "services" "xserver" "startGnuPGAgent" ] ''
+      GnuPG 2.1.x changed the way the gpg-agent works, and that new approach no
+      longer requires (or even supports) the "start everything as a child of the
+      agent" scheme we've implemented in NixOS for older versions.
+
+      To configure the gpg-agent for your X session, add the following code to
+      ~/.bashrc or some file that’s sourced when your shell is started:
+
+          GPG_TTY=$(tty)
+          export GPG_TTY
+
+      If you want to use gpg-agent for SSH, too, also add the settings
+
+          gpg-connect-agent /bye
+          unset SSH_AGENT_PID
+          export SSH_AUTH_SOCK="''${HOME}/.gnupg/S.gpg-agent.ssh"
+
+      and make sure that
+
+          enable-ssh-support
+
+      is included in your ~/.gnupg/gpg-agent.conf.
+      You will need to use `ssh-add` to re-add your ssh keys.
+
+      If gpg’s automatic transformation of the private keys to the new format fails,
+      you will need to re-import your private keyring as well:
+
+          gpg --import ~/.gnupg/secring.gpg
```

Comment:
Huh? Why would it be necessary to import the secret key ring into itself? That sounds like a no-op to me. If that is really necessary, then this is clearly a bug in GnuPG that should be reported upstream so that it can be fixed.